website/files/models/osfstorage.py

```diff
@@ -82,9 +83,39 @@ def move_under(self, destination_parent, name=None):
         return super(OsfStorageFileNode, self).move_under(destination_parent, name)
 
     def save(self):
+        """
+        Saves node, logs any check-out or check-in actions
+            but we want to log specific activities (checkout, checkin)
+            check that via the if in the middle of the method and log the correct activity
+
+        :return:
+        """
         self.path = ''
         self.materialized_path = ''
-        return super(OsfStorageFileNode, self).save()
+        ret = super(OsfStorageFileNode, self).save()
+
+        if 'checkout' in ret and len(ret) == 1:
```

Comment:
Slightly more complicated than that -- the `checkout` field is being updated in APIv2, this just detects it and logs accordingly. That being said, it could be done there by 1) detecting that a checkout is occurring and 2) calling a new model method that adds the log. Something like:
```python
#api/files/serializer.py:FileSerializer
def update(self, instance, validated_data):
    assert isinstance(instance, FileNode), 'Instance must be a FileNode'
+   user = validated_data.get('checkout', None) or instance.checkout
    for attr, value in validated_data.items():
        setattr(instance, attr, value)
    instance.save()
+   if user and len(validated_data) == 1:
+       instance.log_check_out(user)
    return instance

#website/models/files/osfstorage.py:OsfStorageFileNode
def log_check_out(self, user):
    # logic moved from .save
    self.node.add_log(
        action=['checked_in', 'checked_out'][self.is_checked_out],
        params={
            'project': self.node.parent_id,
            'node': self.node._id,
            'urls': {
                'download': "/project/" + self.node._id + "/files/osfstorage/" + self._id + "/?action=download",
                'view': "/project/" + self.node._id + "/files/osfstorage/" + self._id},
            'path': self.materialized_path
        },
        auth=Auth(user),
    )
```